src/main/scala/com.socrata.geoexport/http/ExportService.scala

```diff
@@ -122,13 +127,79 @@ class ExportService(sodaClient: UnmanagedCuratedServiceClient) extends SimpleRes
     }
   }
 
+  private def parseQueryBody(req: HttpRequest, baseUids: Seq[String]): Either[HttpResponse, Seq[LayerWithQuery]] = {
```

Comment:
This method is information-dense and has me wondering if it could be refined for readability...

Reply:
This might even be *more* information dense, but I tend to find for comprehensions more readable:

```scala
  private def parseQueryBody(req: HttpRequest, baseUids: Seq[String]): Either[HttpResponse, Seq[LayerWithQuery]] = {
    for {
      reader <- req.reader.left.map(_ =>
        BadRequest ~> Json(JsonUtil.renderJson(Map(errorKey -> "Invalid or unparsable Content-Type or MimeType")))
      ).right
      queryBody <- JsonUtil.readJson[QueryBody](reader).left.map(_ =>
        BadRequest ~> Json(JsonUtil.renderJson(Map(errorKey -> s"""Could not parse the QUERY body""")))
      ).right
      layersWithQueries <- JsonUtil.parseJson[Seq[LayerWithQuery]](queryBody.layersWithQueries).left.map(_ =>
        BadRequest ~> Json(JsonUtil.renderJson(Map(errorKey -> s"""Could not parse the layersWithQuerys parameter""")))
      ).right
    } yield {
      baseUids.map(fxf => LayerWithQuery(fxf, queryBody.query, queryBody.context, queryBody.copy)) ++ layersWithQueries
    }
  }
```

Reply:
I ended up modifying how the query body is passed so that LayersWithQueries gets passed as regular json and not a string, and thus doesn't need to be parsed out separately! That leaves it with just two layers of Right/Left matching, which I find easier to read in this format than a for comprehension, if that's okay?